uploader.go

```diff
@@ -5,11 +5,29 @@ import (
 )
 
 type Uploader struct {
-	client  *Client
-	url     string
-	upload  *Upload
-	offset  int64
-	aborted bool
+	client   *Client
+	url      string
+	upload   *Upload
+	offset   int64
+	aborted  bool
+	infoSubs []chan UploadInfo
+}
+
+type UploadInfo struct {
```

Comment:
We have an Upload class to hold all information about the upload. Upload class already has the actual file size, it just needs to have the actual offset. If we have both we can calculate the percent completed and if it's 100% it's finished. A